### PR TITLE
mon,mgr: fix "ceph osd df", add some tools to find untested commands

### PR DESCRIPTION
**Pull request by gregsfortytwo**
Fixes http://tracker.ceph.com/issues/20256

This PR moves the "osd df" command handling from the monitor (where it currently crashes out!) to the manager. Along the way, I move the PGStatService into PGMap and the OSDUtilizationDumper code from OSDMonitor into OSDMap.

There's also a little script which greps for COMMANDs and does some basic grepping to see if they are unused in the qa and src/test directories, so we can identify them in future.

### CODE REVIEW

**Comment by gregsfortytwo** [comment]
There are several commands which fail the check_commands.sh checks right now; I will create tickets for them in the appropriate projects.
I also didn't actually add one for "osd df"; I'll push one in a moment. Are we interested in trying to make check_commands.sh part of the "make check" tests?

**Comment by dmick** [comment]
it kind of is, with cephtool-test-X.sh ?

**Comment by gregsfortytwo** [comment]
@dmick, I notice an arm build failure here which seems to be a build infrastructure program rather than a code issue.

File: src/mon/MgrMonitor.h [hunk 1]
```diff
@@ -18,7 +18,7 @@
 #include "MgrMap.h"
 #include "PaxosService.h"
 
-class PGStatService;
+class MonPGStatService;
```
Comment (gregsfortytwo) [inline src/mon/MgrMonitor.h]: Done.

**Comment by gregsfortytwo** [comment]
Added a ticket for check_commands.sh once it's cleaned up.
http://tracker.ceph.com/issues/20300

**Comment by gregsfortytwo** [comment]
And yeah, sorry about the obtuse naming of all the StatService derivatives. :( I couldn't come up with anything better and figure it won't matter very often.

**Comment by liewegas** [comment]
I don't really follow what the *Services (there are more than one now?) are for or why they vary, but they'll presumably all get removed right after luminous branches off so I don't have any strong feelings about it!  Otherwise this looks good to me!

**Comment by gregsfortytwo** [comment]
> I don't really follow what the *Services (there are more than one now?) are for or why they vary

The osd dumping stuff takes a PGStatService, so I moved the base class into PGMap and have an implementation running on the manager. Then the others inherit from that in various ways (and there's a monitor-specific base class adding its necessary functionality). It's tedious but whatever.

**Comment by gregsfortytwo** [comment]
I'll squash this once we are happy with testing? :)

**Comment by yuriw** [comment]
@gregsfortytwo seems having conflict

Auto-merging src/osd/OSDMap.h
Auto-merging src/osd/OSDMap.cc
CONFLICT (content): Merge conflict in src/osd/OSDMap.cc
Auto-merging src/mon/OSDMonitor.cc
Auto-merging src/mon/MonCommands.h
Auto-merging qa/workunits/cephtool/test.sh
Automatic merge failed; fix conflicts and then commit the result.

**Comment by gregsfortytwo** [comment]
Rebased and squashed @yuriw

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-06-15_16:44:51-rados-wip-sage-testing-distro-basic-smithi/

**Pull request merged**